base/shell/explorer/traywnd.cpp

```diff
@@ -99,6 +99,23 @@ VOID RestoreWindowPos()
     g_WindowPosBackup.RemoveAll();
 }
 
+BOOL HasToBeMinimized(HWND hwnd)
+{
+    if (::IsWindowVisible(hwnd) && !::IsIconic(hwnd) && ::IsWindowEnabled(hwnd))
+    {
+        if (::GetClassLongPtrW(hwnd, GCW_ATOM) == (ULONG_PTR)WC_DIALOG)
```

Comment:
Oh, so now *all* dialog windows can be minimized? Even those without minimize button or sthg?

Reply:
Visible, non-minimized, and enabled dialog windows can be.

Reply:
But is this because they have a button in the taskbar ?

Reply:
1. Created a window of the combination of `WS_POPUP` window style and `WS_EX_TOOLWINDOW` extended style.
2. It won't be shown on taskbar.
3. But `Win+D` works on it.

Reply:
> 3. But Win+D works on it.

Even if that dialog, with those styles, has its minimize button disabled or hidden?

Reply:
Yes, it is. I did test.

https://jira.reactos.org/secure/attachment/66044/DialogTest.zip